recipes/botan/all/conanfile.py

```diff
@@ -147,49 +153,47 @@ def validate(self):
                 raise ConanInvalidConfiguration('{0} requires non-header-only static boost, without magic_autolink, and with these components: {1}'.format(self.name, ', '.join(self._required_boost_components)))
 
         compiler = self.settings.compiler
-        version = tools.Version(self.settings.compiler.version)
+        compiler_version = Version(self.settings.compiler.version)
 
-        if compiler == 'Visual Studio' and version < '14':
-            raise ConanInvalidConfiguration("Botan doesn't support MSVC < 14")
+        check_min_vs(self, 190)
 
-        elif compiler == 'gcc' and version >= '5' and compiler.libcxx != 'libstdc++11':
+        if compiler == 'gcc' and compiler_version >= "5" and compiler.libcxx != 'libstdc++11':
             raise ConanInvalidConfiguration(
                 'Using Botan with GCC >= 5 on Linux requires "compiler.libcxx=libstdc++11"')
 
-        elif compiler == 'clang' and compiler.libcxx not in ['libstdc++11', 'libc++']:
+        if compiler == 'clang' and compiler.libcxx not in ['libstdc++11', 'libc++']:
             raise ConanInvalidConfiguration(
                 'Using Botan with Clang on Linux requires either "compiler.libcxx=libstdc++11" ' \
                 'or "compiler.libcxx=libc++"')
 
         # Some older compilers cannot handle the amalgamated build anymore
         # See also https://github.com/randombit/botan/issues/2328
-        if tools.Version(self.version) >= '2.14.0' and self.options.amalgamation:
-            if (compiler == 'apple-clang' and version < '10') or \
-               (compiler == 'gcc' and version < '8') or \
-               (compiler == 'clang' and version < '7'):
+        if Version(self.version) >= '2.14.0' and self.options.amalgamation:
+            if (compiler == 'apple-clang' and compiler_version < '10') or \
+               (compiler == 'gcc' and compiler_version < '8') or \
+               (compiler == 'clang' and compiler_version < '7'):
                 raise ConanInvalidConfiguration(
-                    'botan amalgamation is not supported for {}/{}'.format(compiler, version))
+                    'botan amalgamation is not supported for {}/{}'.format(compiler, compiler_version))
 
         if self.options.get_safe("single_amalgamation", False) and not self.options.amalgamation:
             raise ConanInvalidConfiguration("botan:single_amalgamation=True requires botan:amalgamation=True")
 
     def source(self):
-        tools.get(**self.conan_data['sources'][self.version], strip_root=True, destination=self._source_subfolder)
+        get(conanfile=self, **self.conan_data['sources'][self.version], strip_root=True)
 
     def build(self):
-        for patch in self.conan_data.get('patches', {}).get(self.version, []):
-            tools.patch(**patch)
-        with tools.chdir(self._source_subfolder):
+        apply_conandata_patches(self)
```

Comment:
Maybe a Conan expert can answer this with a reason why:
Is `build()` or `source()` the best place to put this?

In my change I moved this to  `source()`, because the Boost recipe calls it there too, and it made sense to me conceptually. But in most recipe it seems to be called in the `build()` method.